contracts/savings/SavingsManager.sol

```diff
@@ -140,32 +180,98 @@ contract SavingsManager is ISavingsManager, PausableModule {
      */
     function depositLiquidation(address _mAsset, uint256 _liquidated)
         external
+        whenNotPaused
         onlyLiquidator
+        whenStreamsNotFrozen
     {
         // transfer liquidated mUSD to here
         IERC20(_mAsset).safeTransferFrom(_liquidator(), address(this), _liquidated);
 
+        uint256 leftover = _allUnclaimedRewards(_mAsset);
+        _initialiseStream(_mAsset, _liquidated.add(leftover), DURATION);
+
+        emit LiquidatorDeposited(_mAsset, _liquidated);
+    }
+
+    /**
+     * @dev Collects the platform interest from a given mAsset and then adds capital to the
+     * stream. If there is > 24h left in current stream, just top it up, otherwise reset.
+     * @param _mAsset The mAsset to fetch interest
+     */
+    function collectAndStreamInterest(address _mAsset)
+        external
+        whenNotPaused
+        whenStreamsNotFrozen
+    {
+        ISavingsContract savingsContract = savingsContracts[_mAsset];
+        require(address(savingsContract) != address(0), "Must have a valid savings contract");
+
         uint256 currentTime = now;
+        uint256 previousBatch = lastBatchCollected[_mAsset];
+        uint256 timeSincePreviousBatch = currentTime.sub(previousBatch);
+        require(timeSincePreviousBatch > 12 hours, "Cannot deposit twice in 12 hours");
+        lastBatchCollected[_mAsset] = currentTime;
 
-        // Get remaining rewards
-        uint256 end = rewardEnd[_mAsset];
+        // Batch collect
+        IMasset mAsset = IMasset(_mAsset);
+        (uint256 interestCollected, uint256 totalSupply) = mAsset.collectPlatformInterest();
+
+        if(interestCollected > 0){
+            // Validate APY
+            uint256 apy = _validateCollection(totalSupply, interestCollected, timeSincePreviousBatch);
+
+            // Get remaining rewards
+            uint256 leftover = _allUnclaimedRewards(_mAsset);
+
+            // Bundle this yield in with liquidator yield
+            // If (remainingTime < 24h)
+            //       take all remaining, add this, stream over next 24h
+            // else stream over remainingTime
+            uint256 end = streamEnd[_mAsset];
+            uint256 remaining = end > currentTime ? end.sub(currentTime) : 0;
+            uint256 newDuration = remaining < ONE_DAY ? ONE_DAY : remaining;
```

Comment:
It seems like there are two different durations that share the same stream: 1 day and 1 week. This seems to complicate determining the new duration slightly. For instance, there might be 6 days remaining. Now, when a 1 day stream is added the yield is bundled with the existing one and streamed over a longer period.

Have you considered using to separate streams?

Reply:
This was done primarily to reduce gas costs as much as possible. The goal was to share the `lastCollection` property.

I will action this by splitting out into two streams, and if possible, continuing to share the `lastCollection`. If not, then I will introduce a second timestamp for the platform interest yield.

Sharing the same collection time = 2x SLOAD 
Having a separate collection time. = 3x SLOAD and 1x SSTORE

Reply:
Created two seperate streams, both calling _collectAndDistributeInterest before they are set. This causes the lastCollected timestamp to always be set correctly.